File: sdk/examples/java/src/main/java/io/lakehold/examples/StreamQuery.java
package io.lakehold.examples;

import io.lakehold.sdk.ApiClient;
import io.lakehold.sdk.runtime.LakeholdRuntime;
import java.time.Duration;

public final class StreamQuery {
    private StreamQuery() { }

    public static void main(String[] args) throws Exception {
        ApiClient client = LakeholdRuntime.configure(
            new ApiClient().setBasePath(required("LAKEHOLD_URL")), Duration.ofSeconds(30));
        LakeholdRuntime.streamQuery(client, required("LAKEHOLD_TOKEN"),
            required("LAKEHOLD_TENANT"), required("LAKEHOLD_CATALOG"), "SELECT 1 AS value",
            event -> System.out.println(event.payload()));
    }

    private static String required(String name) {
        String value = System.getenv(name);
        if (value == null || value.isEmpty()) {
            throw new IllegalStateException(name + " is required");
        }
        return value;
    }
}

File: sdk/examples/python/stream_query.py
"""Stream one query using the source checkout."""

import os

import lakehold_sdk
from lakehold_sdk.runtime import LakeholdApiClient, stream_query


def required(name: str) -> str:
    value = os.getenv(name)
    if not value:
        raise RuntimeError(f"{name} is required")
    return value


configuration = lakehold_sdk.Configuration(
    host=required("LAKEHOLD_URL"),
    access_token=required("LAKEHOLD_TOKEN"),
)
with LakeholdApiClient(configuration, timeout=30) as client:
    for event in stream_query(
        client,
        tenant=required("LAKEHOLD_TENANT"),
        catalog=required("LAKEHOLD_CATALOG"),
        sql="SELECT 1 AS value",
    ):
        print(event.payload)
